code/modules/mob/living/carbon/carbon_update_icons.dm

```diff
@@ -266,12 +282,52 @@
 		else
 			new_limbs += limb_icon_cache[icon_render_keys[limb.body_zone]] //Pulls existing sprites from the cache
 
+	remove_overlay(BODYPARTS_LAYER)
+
 	if(new_limbs.len)
 		overlays_standing[BODYPARTS_LAYER] = new_limbs
 
 	apply_overlay(BODYPARTS_LAYER)
 
 
+/**
+ * Here we update the legs separately from the other bodyparts. Thanks BYOND for so little support for dir layering.
+ *
+ * Arguments:
+ * * right_leg - Right leg that we might need to update. Can be null.
+ * * left_leg - Left leg that we might need to update. Can be null.
+ * * old_left_leg_key - The icon_key of the left_leg, passed here to avoid having to re-generate it in this proc.
+ *
+ * Returns a boolean, TRUE if the legs need to be redrawn, FALSE if they do not need to be redrawn.
+ * Necessary so that we can ensure that modifications of legs cause overlay updates.
+ */
+/mob/living/carbon/proc/update_legs(obj/item/bodypart/r_leg/right_leg, obj/item/bodypart/l_leg/left_leg, old_left_leg_key)
+	var/list/left_leg_icons // yes it's actually a list, bet you didn't expect that, now did you?
+	var/legs_need_redrawn = FALSE
+	if(left_leg)
+		// We regenerate the look of the left leg if it isn't already cached, we don't if not.
+		if(!(icon_render_keys[left_leg.body_zone] == old_left_leg_key))
+			limb_icon_cache[icon_render_keys[left_leg.body_zone]] = left_leg.get_limb_icon()
+			legs_need_redrawn = TRUE
+
+		left_leg_icons = limb_icon_cache[icon_render_keys[left_leg.body_zone]]
+
+	if(right_leg)
+		var/old_right_leg_key = icon_render_keys?[right_leg.body_zone]
+		var/left_leg_mask_key = left_leg?.generate_mask_key().Join() // We generate a new mask key, to see if it changed.
+		right_leg.left_leg_mask_key = left_leg ? left_leg_mask_key : null
+		// We regenerate the left_leg_mask in case that it doesn't exist yet.
+		if(left_leg_mask_key && !right_leg.left_leg_mask_cache[left_leg_mask_key] && left_leg_icons)
+			right_leg.left_leg_mask_cache[left_leg_mask_key] = generate_left_leg_mask(left_leg_icons[1], left_leg_mask_key)
+		// We generate a new icon_render_key, which also takes into account the left_leg_mask_key so we cache the masked versions of the limbs too.
+		icon_render_keys[right_leg.body_zone] = right_leg.is_husked ? right_leg.generate_husk_key().Join() : right_leg.generate_icon_key().Join()
+
```

Comment:
actually, I get the feeling this code would benifit from being made into a proc on the leg, to reduce the amount of thing.x you need to do
also could make the var names shorter, though I'm not sure how exactly you'd do that

Reply:
I personally prefer having slightly longer names than ambiguous ones.

Also, yes, I *could* do that, I didn't in this case because I wanted to leave it all in one place, so that it would be easy to find in case someone wanted to change behavior in there.

Reply:
I'd prefer `r_leg/l_leg` because it makes the code easier to read

Reply:
> I'd prefer `r_leg/l_leg` because it makes the code easier to read

I'd respectfully disagree, the difference between `r` and `l` can be tough to spot at a quick glance (especially when you've got attention problems), I dislike using abbreviations when I can write out the full word, it's easier to spot and to search for that way, in my personal opinion.